clients/src/main/java/org/apache/kafka/common/network/ChannelBuilders.java

```diff
@@ -48,30 +53,42 @@ public static ChannelBuilder clientChannelBuilder(SecurityProtocol securityProto
             if (clientSaslMechanism == null)
                 throw new IllegalArgumentException("`clientSaslMechanism` must be non-null in client mode if `securityProtocol` is `" + securityProtocol + "`");
         }
-        return create(securityProtocol, Mode.CLIENT, loginType, configs, clientSaslMechanism, saslHandshakeRequestEnable, null);
+        return create(securityProtocol, Mode.CLIENT, loginType, config, listenerName, clientSaslMechanism,
+                saslHandshakeRequestEnable, null);
     }
 
     /**
+     * @param listenerName the listenerName
      * @param securityProtocol the securityProtocol
-     * @param configs server configs
+     * @param config server config
      * @param credentialCache Credential cache for SASL/SCRAM if SCRAM is enabled
      * @return the configured `ChannelBuilder`
      */
-    public static ChannelBuilder serverChannelBuilder(SecurityProtocol securityProtocol,
-            Map<String, ?> configs,
-            CredentialCache credentialCache) {
-        return create(securityProtocol, Mode.SERVER, LoginType.SERVER, configs, null, true, credentialCache);
+    public static ChannelBuilder serverChannelBuilder(ListenerName listenerName,
+                                                      SecurityProtocol securityProtocol,
+                                                      AbstractConfig config,
+                                                      CredentialCache credentialCache) {
+        return create(securityProtocol, Mode.SERVER, JaasContext.Type.SERVER, config, listenerName, null,
+                true, credentialCache);
     }
 
     private static ChannelBuilder create(SecurityProtocol securityProtocol,
                                         Mode mode,
-                                        LoginType loginType,
-                                        Map<String, ?> configs,
+                                        JaasContext.Type loginType,
+                                        AbstractConfig config,
+                                        ListenerName listenerName,
                                         String clientSaslMechanism,
                                         boolean saslHandshakeRequestEnable,
                                         CredentialCache credentialCache) {
-        ChannelBuilder channelBuilder;
+        Map<String, ?> configs;
+        if (listenerName == null)
+            configs = config.values();
+        else {
+            String prefix = "listener.name." + listenerName.value().toLowerCase(Locale.ROOT) + ".";
```

Comment:
It will be good to define a constant for `listener.name` prefix. Perhaps the prefixing could be moved to `ListenerName`?